nvtabular/ops/base.py

```diff
@@ -100,10 +99,46 @@ def label(self) -> str:
 
     @property
     def supports(self) -> Supports:
-        """ Returns what kind of data representation this operator supports """
+        """Returns what kind of data representation this operator supports"""
         return Supports.CPU_DATAFRAME | Supports.GPU_DATAFRAME
 
-    def inference_initialize(self, columns: ColumnNames, model_config: dict) -> Optional[Operator]:
+    def inference_initialize(self, columns: Columns, model_config: dict) -> Optional[Operator]:
         """Configures this operator for use in inference. May return a different operator to use
         instead of the one configured for use during training"""
         return None
+
+
+class AddMetadata(Operator):
+    def __init__(
+        self,
+        tags=None,
+        properties=None,
+        is_target=False,
+        is_regression_target=False,
+        is_binary_target=False,
+        is_multi_class_target=False,
+    ):
+        super().__init__()
+        self.properties = properties or {}
+        if isinstance(tags, DefaultTags):
+            tags = tags.value
+        if not tags:
+            tags = []
+        if not isinstance(tags, list):
+            tags = [tags]
+        if is_target:
+            tags.extend(DefaultTags.TARGETS.value)
+        if is_regression_target:
+            tags.extend(DefaultTags.TARGETS_REGRESSION.value)
+        if is_multi_class_target:
+            tags.extend(DefaultTags.TARGETS_MULTI_CLASS.value)
+        if is_binary_target:
+            tags.extend(DefaultTags.TARGETS_BINARY.value)
+
+        self.tags = list(set(tags))
+
+    def transform(self, columns: Columns, df: DataFrameType) -> DataFrameType:
+        return df
+
+    def output_columns(self, columns: Columns) -> Columns:
+        return columns.map(lambda col: col.with_tags(self.tags).with_properties(**self.properties))
```

Comment:
If we were to merge those _with calls... we would only need to return the columnGroup once... If I am reading this correctly we will first create a columngroup with tags then another columngroup with those tags and the properties...

Reply:
Yeah, I think you're right. Since we're not using the properties yet, I'd be happy just to remove `with_properties` for now and kick the can down the road a little bit.